Source/WEB_ROOT/Inventory/js/editAsset.js

```diff
@@ -0,0 +1,206 @@
+	    addRecordToHistoryDB("~(gpv.assetid)","Asset was Edited from the Edit Asset page by ~[x:username]");
+	});
+});
+
+
+// OnChange Events
+$j('#schoolDropDown').change(function(){
+		//The school drop down has changed
+		SetDropDowns();
+        $j('#storeSchoolId').val($j('#schoolDropDown :selected').val());
+        $j('#storeSchoolName').val($j('#schoolDropDown :selected').text());
+
+        //show the student / teacher check box
+        showHideFormElements("assignedToElement", true);
+});
+
+function createlisteners()
+{
+	$j('#teachersDropdown').change(function(){
+	    //The Teachers drop down has changed
+		//console.log($j('#teachersDropdown :selected').val());
+		 $j('#AssignedTo_dcid').val($j('#teachersDropdown :selected').val());
+		 $j('#AssignedTo_name').val($j('#teachersDropdown :selected').text());
+		
+		 if($j('#teachersDropdown :selected').val() != "")
+		 {
+			toggleAssetElements(true);
+		 }else{
+			toggleAssetElements(false);
+		 }
+		 
+	});
+
+	$j('#studentsToDropdown').change(function(){
+	    //The Teachers drop down has changed
+		//console.log($j('#studentsToDropdown :selected').val());
+		$j('#AssignedTo_dcid').val($j('#studentsToDropdown :selected').val());
+		$j('#AssignedTo_name').val($j('#studentsToDropdown :selected').text());
+		if($j('#studentsToDropdown :selected').val() != "")
+		 {
+			toggleAssetElements(true);
+		 }else{
+			toggleAssetElements(false);
+		 }
+	});
+}
+//End Onchange Events
+
+var assignedType;
+
+function SetDropDowns(){
+	//clear the dropdowns
+	document.getElementById('studentsToDropdown').options.length = 0;
+	document.getElementById('teachersDropdown').options.length = 0;
+
+	//run Ajax to get the dcid
+	 
+	if (document.getElementById('isStudentCheckBox').checked) 
+	{
+		assignedType = 1;
+	} else {
+		assignedType = 0;
+	};
+
+	var url = "scripts/AssignedToDropDown.html?type=" + assignedType +"&id=" + "&schoolid=" + $j('#schoolDropDown :selected').val();
+	//console.log(url);
+	$j.ajax(
+	{
+		url: url, 
+		dataType: "text",
+		success: function(result)
+		{
+			
+			if (assignedType == 1) 
+			{
+				//push students to the dropdown
+				var options = '<select id="studentsToDropdown">' + result + '</select>';
+				document.getElementById( 'studentsToDropdown' ).outerHTML = options;
+			} else {
+				//push teachers to the dropdown
+				var options = '<select id="teachersDropdown">' + result + '</select>';
+				document.getElementById( 'teachersDropdown' ).outerHTML = options;
+			};
+			createlisteners();
+		}
+	});
+}
+
+function checkForDup(assetid)
+{
+    var serial = document.getElementById("serialInput").value
+
+    $j.ajax(
+	{
+		url: "scripts/checkDupSerial.html?serial=" + serial + "&assetid=" + assetid, 
+		dataType: "text",
+		success: function(result)
+		{
+            var warning = document.getElementById("serialDupWarning");
+            
+			if(result.length > 0)
+			{
+				warning.style.display = "block";
+			}else
+			{

```

Comment:
Just one of those points where the consistency of your syntax is off just a bit. I won't point it out in the future. Yet it can be helpful to always use }else{ or having them on their own lines. Javascript is a bit of a pain, but most tools like vs and resharper will help you hunt down these little syntax discrepancies. Writing code is much more like writing a novel than we give it credit for. The reader should read top down and understand your intent.